src/robusta/core/sinks/robusta/dal/supabase_dal.py

```diff
@@ -277,6 +279,80 @@ def get_active_services(self) -> List[ServiceInfo]:
             for service in res.get("data")
         ]
 
+    def get_active_nodes(self) -> List[NodeInfo]:
+        res = (
+            self.client.table(NODES_TABLE)
+            .select(
+                "name", "node_creation_time", "taints", "conditions", "memory_capacity", "allocatable_memory",
+                "allocated_memory", "cpu_capacity", "allocatable_cpu", "allocated_cpu", "pods_count", "pods",
+                "internal_ip", "external_ip", "node_info",
+            )
+            .filter("account_id", "eq", self.account_id)
+            .filter("cluster_id", "eq", self.cluster)
+            .filter("deleted", "eq", False)
+            .execute()
+        )
+        if res.get("status_code") not in [200]:
+            msg = f"Failed to get existing nodes (supabase) error: {res.get('data')}"
+            logging.error(msg)
+            self.handle_supabase_error()
+            raise Exception(msg)
+
+        return [
+            NodeInfo(
+                name=node["name"],
+                node_creation_time=node["node_creation_time"],
+                taints=node["taints"],
+                conditions=node["conditions"],
+                memory_capacity_mb=node["memory_capacity"],
+                allocatable_memory_mb=node["allocatable_memory"],
+                allocated_memory_mb=node["allocated_memory"],
+                cpu_capacity=node["cpu_capacity"],
+                allocatable_cpu=node["allocatable_cpu"],
+                allocated_cpu=node["allocated_cpu"],
+                pods_count=node["pods_count"],
+                pods=node["pods"],
+                internal_ip=node["internal_ip"],
+                external_ip=node["external_ip"],
+                node_info=json.loads(node["node_info"]),
+            )
+            for node in res.get("data")
+        ]
+
+    def __to_db_node(self, node: NodeInfo) -> Dict[Any, Any]:
```

Comment:
Here to you can use `self.dict()` if the names are the same.

Reply:
they aren't

Reply:
fixed